### DISCUSSION TITLE
anchor in url does not scroll to id when using wire:navigate

### DISCUSSION THREAD

**Discussion by bezanis** [livewire/livewire#6692] [7. Bugs (archived)]
### Livewire version

3.0.1

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Normally when navigating to a link containing an #anchor, the page loads scrolled to the position of #anchor.
When navigating to a url containing an #anchor combined with wire:navigate, the page loads but is not scrolled to the position of the id. 
In this example the first link loads the page scrolled to the <p> with the id 'starthere'.
However, the second link loads the page but not scrolled to the id 'starthere'.
`<br/>
    <a href="/curpage#starthere">Navigate without wire:navigate</a><br/>
    <a href="/curpage#starthere" wire:navigate>Navigate using wire:navigate</a>
    <div class="h-[3000px]">
        Filler
    </div>
    <p id="starthere">Start here</p>
    <div class="h-[3000px]">
        More Filler
    </div>`

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by xaimes** [comment]
Same. Does anyone have a solution?

**Reply by bezanis** [reply]
The workaround I've made is to set a 1ms timeout to wait for the redraw, and then check for the hash

window.addEventListener('livewire:navigated', () => {
  setTimeout(() => {
    if (window.location.hash == "") {
      window.scrollTo({top: 0});
    } else {
      const element = document.querySelector(window.location.hash)
      element.scrollIntoView({
        behavior: 'smooth'
      })
    }
  }, 1)
})

In this example, it's using a smooth scroll, but you could use an instant scroll if you don't like the effect

**Reply by l3eto** [reply]
this make an effect very bad, goes top, then scroll, if you are doing this on the same page...

**Reply by danie-ramdhani** [reply]
FYI, `wire:navigate` url with anchor tag/hash tag already fixed in https://github.com/livewire/livewire/pull/8143

**Comment by serge-kilimoff** [comment]
Hello,

The bug on anchors did not appear on version 3.5.4. On version 3.5.9 (I updated my code yesterday), the reference to an anchor does not work.

Therefore, do you have to stay stuck in 3.5.4 to have this basic functionality?

Thank you.

**Reply by danie-ramdhani** [reply]
read https://github.com/livewire/livewire/pull/8143#issuecomment-2391190321 and https://github.com/livewire/livewire/pull/8883

**Reply by serge-kilimoff** [reply]
3.5.4 or 3.5.6, , it doesn't matter. The bottom line is if there is a solution with the latest stable version of Livewire. And of course, the point of my post is to emphasize that there is an error, as there is no more ticket management on this project.
